relay-server/src/actors/processor.rs

```diff
@@ -59,12 +59,42 @@ use crate::envelope::{AttachmentType, ContentType, Envelope, Item, ItemType};
 use crate::extractors::RequestMeta;
 use crate::metrics_extraction::sessions::extract_session_metrics;
 use crate::metrics_extraction::transactions::extract_transaction_metrics;
-use crate::metrics_extraction::transactions::types::ExtractMetricsError;
+use crate::metrics_extraction::transactions::types::CommonTags;
+use crate::metrics_extraction::transactions::types::{
+    ExtractMetricsError, TransactionMeasurementTags, TransactionMetric,
+};
+use crate::metrics_extraction::IntoMetric;
 use crate::statsd::{RelayCounters, RelayHistograms, RelayTimers};
 use crate::utils::{
     self, get_sampling_key, log_transaction_name_metrics, ChunkedFormDataAggregator, FormDataIter,
     ItemAction, ManagedEnvelope, SamplingResult,
 };
+use once_cell::sync::Lazy;
+
+/// In 'fn process_measurements' we want to check if a measurement name is too long, to do that we
+/// check with a maximum length provided by the config, but that length represents the entire MRI name,
+/// and in process_measurements we only have access to the name part, and the unit part, so here
+/// we calculate the length of the remainder part. which isn't changing so we only need to do it once.
+static FIXED_MEASUREMENT_LEN: Lazy<usize> = Lazy::new(|| {
```

Comment:
It's not for testing, but i agree it's confusing (thats why i made big description). Basically the value we use to check if a metric name is too long actually represents the entire MRI string, but in the process function we only have access to the name and the unit, you gotta also know how big the other parts are, so this function basically tries to compute the size of MRI without unit and name for measurements.

i could have hardcoded it but that would make maintaining harder in case we change something

Reply:
I'd prefer to have this in the same file as `process_measurements` but for that i might have to move a lot of stuff around as it has dependencies on `relay-server` which can't be a dependency of `relay-general`